pkgs/test/top-level/stage.nix

```
@@ -0,0 +1,116 @@
+# run like this:
+#  nix-build pkgs/test/top-level/stage.nix
+{
+  localSystem ? {
+    system = builtins.currentSystem;
+  },
+}:
+
+with import ../../top-level { inherit localSystem; };
+
+let
+  # To silence platform specific evaluation errors
+  discardEvaluationErrors =
+    e:
+    let
+      res = builtins.tryEval e;
+    in
+    res.success == res.value;
+
+  # Basic test for idempotency of the package set, i.e:
+  # Applying the same package set twice should work and
+  # not change anything.
+  isIdempotent = set: discardEvaluationErrors (pkgs.${set}.stdenv == pkgs.${set}.${set}.stdenv);
+
+  # Some package sets should be noops in certain circumstances.
+  # This is very similar to the idempotency test, but not going
+  # via the super' overlay.
+  isNoop =
+    parent: child:
+    discardEvaluationErrors (
+      (lib.getAttrFromPath parent pkgs).stdenv == (lib.getAttrFromPath parent pkgs).${child}.stdenv
+    );
+
+  allMuslExamples = builtins.attrNames (
+    lib.filterAttrs (_: { config, ... }: lib.hasSuffix "-musl" config) lib.systems.examples
+  );
+
+  allLLVMExamples = builtins.attrNames (
+    lib.filterAttrs (
+      _:
+      {
+        useLLVM ? false,
+        ...
+      }:
+      useLLVM
+    ) lib.systems.examples
```

```suggestion
    lib.filterAttrs (_: system: system.useLLVM or false) lib.systems.examples
```

More readable than argument destructuring when auto-formatting with nixfmt.